pkg/plugins/middlewares.go

```diff
@@ -2,41 +2,53 @@ package plugins
 
 import (
 	"context"
+	"encoding/json"
 	"fmt"
 	"net/http"
+	"os"
 	"path"
 	"reflect"
 	"strings"
 
+	"github.com/http-wasm/http-wasm-host-go/handler"
+	wasm "github.com/http-wasm/http-wasm-host-go/handler/nethttp"
 	"github.com/mitchellh/mapstructure"
+	"github.com/traefik/traefik/v3/pkg/middlewares"
 	"github.com/traefik/yaegi/interp"
 )
 
 // Build builds a middleware plugin.
 func (b Builder) Build(pName string, config map[string]interface{}, middlewareName string) (Constructor, error) {
-	if b.middlewareBuilders == nil {
-		return nil, fmt.Errorf("no plugin definition in the static configuration: %s", pName)
+	if b.yaegiMiddlewareBuilders == nil || b.wasmMiddlewareBuilders == nil {
+		return nil, fmt.Errorf("no plugin definitions in the static configuration: %s", pName)
 	}
 
-	descriptor, ok := b.middlewareBuilders[pName]
+	// plugin (pName) can be located in yaegi or wasm middleware builders
```

Comment:
@juliens the logic is now divided to yaegiMiddlewareBuilders and wasmMiddlewareBuilders. All other functionality is in separate functions and classes but this Build function will build middlewares for wasm and yaegi. The main problem in this is that if someone has similar named yaegi and wasm plugin, it will always use yaegi. However, I do not see this quite big issue?

Reply:
hmm actually, that is not perhaps possible because the configurations are coming from yaml. It does not allow duplicate keys. It will load only one of them anyway

Reply:
And I think that the problem already exists with two yaegi plugin.